src/components/Board.js

```diff
@@ -5,28 +5,130 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
+
+
+
+
+const CARDS_URL = 'https://inspiration-board.herokuapp.com/boards/brenda/cards'
+
 
 class Board extends Component {
+
+
   constructor() {
     super();
 
     this.state = {
       cards: [],
+      status: {
+        message: "Successfuly cards loaded!",
+        type: "success"
+      }
     };
   }
 
-  render() {
-    return (
-      <div>
-        Board
-      </div>
-    )
+
+  componentDidMount = () => {
+
+    let board_name = this.props.boardName
+    axios.get('https://inspiration-board.herokuapp.com/boards/brenda/cards')
+
+    .then( (response) => {
+
+      this.props.updateStatusCallback(`Successfully loaded  cards from ${board_name}!`, 'success');
+
+      const cardsData = response.data.map((card) =>{
+        return {text: card.card.text,
+          emoji: card.card.emoji,
+          id: card.card.id
+        };
+      });
+      this.setState({cards: cardsData})
+    })
+    .catch((error) => {
+      this.props.updateStatusCallback(error.message, 'error');
+    })
   }
 
+
+  deleteCard = (id) => {
+
+    let board_name = this.props.boardName;
+
+    this.props.updateStatusCallback(`Deleting card ${id}`, 'success');
+
+    axios.delete(`https://inspiration-board.herokuapp.com/boards/${this.props.boardName}/cards/${id}`)
+    .then((response) => {
+
+      this.props.updateStatusCallback(`Successfully deleted card ${id} from ${board_name}´s board!`, 'success');
+
+      const updatedCardsList = []
+      this.state.cards.map((card) => {
+
+        if (card.id !== id) {
+          updatedCardsList.push(card);
+        }
+
+      });
+      this.setState({cards: updatedCardsList});
+
+    })
+    .catch( (error) => {
+      console.log(error);
+
+      this.props.updateStatusCallback(error.message, 'success');
+    });
+  }
+
+
+  addCard = (card) => {
+
+    let updatedCards = this.state.cards;
+
+    axios.post(CARDS_URL, card)
+    .then((response) => {
+
+      this.props.updateStatusCallback(`Sucessfully added a new card!` , 'success');
+
+
+      card.id = response.data.card.id;
+      updatedCards.push(card);
```

Comment:
Good use of closures to easily re-use the card data from the form (in the `card` variable), to later on add that data into state even though the API did not return a response which included all of the card data.